tests/store/test_file_store.py

```diff
@@ -288,32 +288,38 @@ def test_list_run_infos(self):
                 dict_run_info['lifecycle_stage'] = LifecycleStage.ACTIVE
                 self.assertEqual(dict_run_info, dict(run_info))
 
-    def test_log_metric_allows_multiple_values_at_same_ts_and_run_data_uses_max_ts_value(self):
+    def test_log_metric_allows_multiple_values_at_same_step_and_run_data_uses_max_step_value(self):
         fs = FileStore(self.test_root)
         run_uuid = self._create_run(fs).info.run_uuid
 
         metric_name = "test-metric-1"
-        timestamp_values_mapping = {
-            1000: [float(i) for i in range(-20, 20)],
-            2000: [float(i) for i in range(-10, 10)],
-        }
-
-        logged_values = []
-        for timestamp, value_range in timestamp_values_mapping.items():
-            for value in reversed(value_range):
-                fs.log_metric(run_uuid, Metric(metric_name, value, timestamp))
-                logged_values.append(value)
-
-        six.assertCountEqual(
-            self,
-            [metric.value for metric in fs.get_metric_history(run_uuid, metric_name)],
-            logged_values)
-
-        run_metrics = fs.get_run(run_uuid).data.metrics
+        # Check that we get the max of (step, timestamp, value) in that order
+        tuples_to_log = [
+            (0, 100, 1000),
+            (3, 40, 100),  # larger step wins even though it has smaller value
+            (3, 50, 10),  # larger timestamp wins even though it has smaller value
+            (3, 50, 20),  # tiebreak by max value
+            (3, 50, 20),  # duplicate metrics with same (step, timestamp, value) are ok
+            # verify that we can log steps out of order / negative steps
+            (-3, 900, 900),
+            (-1, 800, 800),
+        ]
+        for step, timestamp, value in reversed(tuples_to_log):
```

Comment:
Why reverse?

Reply:
Ah initially I didn't have the  (-3, 900, 900) and (-1, 800, 800) tuples, so the `reversed` was necessary to make sure we didn't just return the last logged value. With the new tuples I think `reversed` is unnecsesary, will remove